sky/clouds/cudo.py

```diff
@@ -0,0 +1,354 @@
+"""Cudo Compute"""
+import json
+import subprocess
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple
+
+from sky import clouds
+from sky import status_lib
+from sky.clouds import service_catalog
+from sky.utils import common_utils
+
+if typing.TYPE_CHECKING:
+    # Renaming to avoid shadowing variables.
+    from sky import resources as resources_lib
+
+_CREDENTIAL_FILES = [
+    # credential files for Cudo,
+    'cudo.yml'
+]
+
+
+def _run_output(cmd):
+    proc = subprocess.run(cmd,
+                          shell=True,
+                          check=True,
+                          stderr=subprocess.PIPE,
+                          stdout=subprocess.PIPE)
+    return proc.stdout.decode('ascii')
+
+
+@clouds.CLOUD_REGISTRY.register
+class Cudo(clouds.Cloud):
+    """Cudo Compute"""
+    _REPR = 'Cudo'
+
+    _INDENT_PREFIX = '    '
+    _DEPENDENCY_HINT = (
+        'Cudo tools are not installed. Run the following commands:\n'
+        f'{_INDENT_PREFIX}  $ pip install cudo-compute')
+
+    _CREDENTIAL_HINT = (
+        'Install cudoctl and run the following commands:\n'
+        f'{_INDENT_PREFIX}  $ cudoctl init\n'
+        f'{_INDENT_PREFIX}For more info: '
+        # pylint: disable=line-too-long
+        'https://skypilot.readthedocs.io/en/latest/getting-started/installation.html'
+    )
+
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.STOP: 'Stopping not supported.',
+        clouds.CloudImplementationFeatures.SPOT_INSTANCE: 'Spot is not supported, as runpod API does not implement spot .',
+        clouds.CloudImplementationFeatures.CUSTOM_DISK_TIER: 'Custom disk tier is currently not supported on Cudo Compute',
+        clouds.CloudImplementationFeatures.DOCKER_IMAGE:
+            ('Docker image is currently not supported on Cudo. '
+             'You can try running docker command inside the '
+             '`run` section in task.yaml.'),
+    }
+    _MAX_CLUSTER_NAME_LEN_LIMIT = 60
+
+    PROVISIONER_VERSION = clouds.ProvisionerVersion.SKYPILOT
+    STATUS_VERSION = clouds.StatusVersion.SKYPILOT
```

Comment:
Since we have this version set to `SKYPILOT`, we need to move the instance status query to the provisioner API, i.e. `sky/provision/cudo/instance.py`